kt/godot-runtime/src/main/kotlin/godot/core/Dictionary.kt

```diff
@@ -0,0 +1,335 @@
+        erase(key)
+        return ret
+    }
+
+    /**
+     * Returns the list of values in the Dictionary.
+     */
+    fun values(): VariantArray<V> {
+        Bridge.engine_call_values(_handle)
+        return (TransferContext.readReturnValue(VariantType.ARRAY) as VariantArray<V>).also {
+            it.variantType = valueVariantType
+        }
+    }
+
+
+    //UTILITIES
+    override operator fun get(key: K): V {
+        TransferContext.writeArguments(keyVariantType to key)
+        Bridge.engine_call_operator_get(_handle)
+        return TransferContext.readReturnValue(valueVariantType, true) as V
+    }
+
+    operator fun set(key: K, value: V) {
+        TransferContext.writeArguments(keyVariantType to key, valueVariantType to value)
+        Bridge.engine_call_operator_set(_handle)
+    }
+
+    operator fun contains(key: K): Boolean = has(key)
+
+    override fun equals(other: Any?): Boolean {
+        if (other == null || other !is Dictionary<*, *>) {
+            return false
+        }
+        TransferContext.writeArguments(VariantType.DICTIONARY to this, VariantType.DICTIONARY to other)
+        Bridge.engine_call_equals(_handle)
+        return TransferContext.readReturnValue(VariantType.BOOL) as Boolean
+    }
+
+    override fun hashCode(): Int {
+        return _handle.hashCode()
+    }
+
+    override fun toString(): String {
+        return "Dictionary($size)"
+    }
+
+    @Suppress("FunctionName")
+    private object Bridge {
+        external fun engine_call_constructor(): VoidPtr
+
+        external fun engine_call_clear(_handle: VoidPtr)
+        external fun engine_call_duplicate(_handle: VoidPtr)
+        external fun engine_call_empty(_handle: VoidPtr)
+        external fun engine_call_erase(_handle: VoidPtr)
+        external fun engine_call_get(_handle: VoidPtr)
+        external fun engine_call_has(_handle: VoidPtr)
+        external fun engine_call_hasAll(_handle: VoidPtr)
+        external fun engine_call_hash(_handle: VoidPtr)
+        external fun engine_call_keys(_handle: VoidPtr)
+        external fun engine_call_size(_handle: VoidPtr)
+        external fun engine_call_values(_handle: VoidPtr)
+        external fun engine_call_operator_get(_handle: VoidPtr)
+        external fun engine_call_operator_set(_handle: VoidPtr)
+        external fun engine_call_equals(_handle: VoidPtr)
+    }
+}
+
+@Suppress("FunctionName")
+inline fun <reified K, reified V> Dictionary(): Dictionary<K, V> {
+    val keyVariantType = variantMapper[K::class]
+    checkNotNull(keyVariantType) {
+        "Can't create a Dictionary with generic key ${K::class}."
+    }
+    val valVariantType = variantMapper[V::class]
+    checkNotNull(valVariantType) {
+        "Can't create a Dictionary with generic value ${V::class}."
+    }
+    return Dictionary<K, V>(keyVariantType, valVariantType)
+}
+
+inline fun <reified K, reified V> dictionaryOf(vararg args: Pair<K, V>) = Dictionary<K, V>().also {
+    it.putAll(args)
+}
```

Comment:
Shouldn't we also have convenience functions to convert a Map into a dictionary other than the corresponding constructor?
Like extension functions for Map itself?

Reply:
yep we can add some I think.

Reply:
done